### ISSUE TITLE
PubMedQA Prompt format.

### ISSUE THREAD

**Issue by badger-lord** [microsoft/BioGPT#86]
It's unclear to me in the paper how the QA is being fed into the model. Not so much context and question, but the answer. 

In the data you have, the long answer is there in all three sets. Does bioGPT use the long answer as context to answer "yes|no|maybe" and the results are the 78.2% reported? Or, does bioGPT score 78.2% by generated the long answer AND answering "yes|no|
maybe"? 

I'm trying to recreate the workflow and having trouble identifying bioGPT's route to this, please advise.

**Comment by ZON-ZONG-MIN** [comment]
The ranking of PubMedQA seems to be unable to find BioGPT.

**Comment by ArvinZhuang** [comment]
@badger-lord what is the expected format? should not include long answer as context for reasoning-required setting right?

**Comment by badger-lord** [comment]
> @badger-lord what is the expected format? should not include long answer as context for reasoning-required setting right?
Pulled from the PubMedQA paper. 
_"A parallel setting, where models can use question and long answer to predict yes/no/maybe answer, is denoted as **reasoning-free setting** since yes/no/maybe are usually explicitly expressed in the long answers"_ 

From the way the training/testing data is set up, I would say BioGPT is reasoning-free, not reasoning-required.

**Comment by ArvinZhuang** [comment]
> > @badger-lord what is the expected format? should not include long answer as context for reasoning-required setting right?
> > Pulled from the PubMedQA paper.
> > _"A parallel setting, where models can use question and long answer to predict yes/no/maybe answer, is denoted as **reasoning-free setting** since yes/no/maybe are usually explicitly expressed in the long answers"_
> 
> From the way the training/testing data is set up, I would say BioGPT is reasoning-free, not reasoning-required.

Yeah, but even not that, reasoning-free should not include context, right?

**Comment by badger-lord** [comment]
> > > @badger-lord what is the expected format? should not include long answer as context for reasoning-required setting right?
> > > Pulled from the PubMedQA paper.
> > > _"A parallel setting, where models can use question and long answer to predict yes/no/maybe answer, is denoted as **reasoning-free setting** since yes/no/maybe are usually explicitly expressed in the long answers"_
> > 
> > 
> > From the way the training/testing data is set up, I would say BioGPT is reasoning-free, not reasoning-required.
> 
> Yeah, but even not that, reasoning-free should not include context, right?

No, reasoning free should include everything, reasoning required should not include long answer. Reasoning free is easier than reasoning required. So, for BioGPT, do you know which they are reporting? I would assume Reasoning free, since their train/test data contains the long answer.

**Comment by ZON-ZONG-MIN** [comment]
@badger-lord  Hi~ Have you tried recreating the workflow?

I follow [BioGPT](https://github.com/microsoft/BioGPT)/[examples](https://github.com/microsoft/BioGPT/tree/main/examples)/[QA-PubMedQA](https://github.com/microsoft/BioGPT/tree/main/examples/QA-PubMedQA)/README.md to Test PubMedQA
`generate_checkpoint.pt` will be generated after running `infer.sh`
I see `generate_checkpoint.pt` does not contain long answer.
I'm not sure what this means?

**Comment by ArvinZhuang** [comment]
@badger-lord I followed [this](https://github.com/microsoft/BioGPT/tree/main/examples/QA-PubMedQA) and seems there is no answer span in their input data. So should be reasoning-required. I got 0.782 with their provided checkpoint.

**Comment by vithursant** [comment]
@badger-lord here 
https://github.com/microsoft/BioGPT/blob/a27f7d5638a7684f29a70a206411ae6786a0f952/examples/QA-PubMedQA/rebuild_data.py#L44 

The `required_long_answer` flag is always False, that would mean that the long answer is not included in the training set. However, in the PubMedQA paper (see Section 3.3 https://aclanthology.org/D19-1259.pdf), the long answer is included in the training but not during the test phase.

Can you confirm this?

**Comment by rezabuughunter** [comment]
Oh my God

**Comment by SantoshGuptaML** [comment]
> @badger-lord I followed [this](https://github.com/microsoft/BioGPT/tree/main/examples/QA-PubMedQA) and seems there is no answer span in their input data. So should be reasoning-required. I got 0.782 with their provided checkpoint.

This was for the large version?

**Comment by ArvinZhuang** [comment]
I think it was Medium size (the one in the paper)

**Comment by SantoshGuptaML** [comment]
> I think it was Medium size (the one in the paper)

The paper had two, in march I think was the update to include the large 1.5 billion parameter model at the end. The initial version I believe only had a 770m parameter model.

**Comment by Steven-Yiran** [comment]
I was able to get a yes/no answer from the `BioGPT-Large-PubMedQA` model with the following prompt:
```
question: <question_text> context <context_text> the answer to the question given the context is
```
as described in the [paper](https://arxiv.org/pdf/2210.10341).
